pkg/analysis/passes/backendbinary/backendbinary.go

```diff
@@ -43,90 +42,88 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	if !ok {
 		return nil, nil
 	}
-	metadataBody, ok := pass.ResultOf[metadata.Analyzer].([]byte)
-	if !ok {
-		return nil, nil
-	}
-
-	var data metadata.Metadata
-	if err := json.Unmarshal(metadataBody, &data); err != nil {
-		return nil, err
-	}
 
-	// alerting declared without backend true
-	if data.Alerting && !data.Backend {
-		pass.ReportResult(
-			pass.AnalyzerName,
-			alertingFoundButBackendFalse,
-			"Found alerting in plugin.json but backend=false",
-			"You have marked your plugin with backend=false in your plugin.json but declared an alerting=true Please set backend=true in your plugin.json if your plugin has a backend component",
-		)
+	metadatamap, ok := pass.ResultOf[nestedmetadata.Analyzer].(nestedmetadata.Metadatamap)
+	if !ok {
 		return nil, nil
 	}
 
-	// executable declared without backend true
-	if data.Executable != "" && !data.Backend {
-		pass.ReportResult(
-			pass.AnalyzerName,
-			backendFoundButNotDeclared,
-			"Found executable in plugin.json but backend=false",
-			"You have marked your plugin with backend=false in your plugin.json but declared a backend executable. Please set backend=true in your plugin.json if your plugin has a backend component or remove the executable from your plugin.json",
-		)
-		return nil, nil
-	}
+	for pluginJsonPath, data := range metadatamap {
```

Comment:
though the changes look big. this is merely moving all the previous logic into a loop of the metadatamap instead of a single metadata